lib/thrift/parser/models.ex

```diff
@@ -469,6 +469,20 @@ defmodule Thrift.Parser.Models do
     defp canonicalize_name(schema, %{name: name} = model) do
       %{model | name: :"#{schema.module}.#{name}"}
     end
+
+    defp put_new_strict(map, key, value) do
+      case map[key] do
+        nil ->
+          Map.put(map, key, value)
+        existing_value ->
+          raise """
+          Name collision!
+          Name:         #{inspect key}
+          First value:  #{inspect value}
+          Second Value: #{inspect existing_value}
```

Comment:
`value` vs `Value`

Also, "First" here is actually the new ("second") value to be inserted. Not sure that really matters, but it could be potentially be clearer.